src/components/PostDetails.tsx

```diff
@@ -1,107 +1,108 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import * as commentsService from '../api/comments';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment as CommentItem } from '../types/Comment';
+import { Post } from '../types/Post';
+import Comment from './Comment';
+
+type Props = {
+  postSelected: Post;
+};
+export const PostDetails: React.FC<Props> = ({ postSelected }) => {
+  const [comments, setComments] = useState<CommentItem[]>([]);
+
+  const [isError, setIsError] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isAddingComment, setIsAddingComment] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    commentsService
+      .getComments(postSelected.id)
+      .then(setComments)
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => setIsLoading(false));
+  }, [postSelected]);
+
+  useEffect(() => {
+    setIsAddingComment(false);
+  }, [postSelected]);
+
+  const addNewComment = useCallback((newComment: CommentItem) => {
+    setComments(prevComments => [...prevComments, newComment]);
+  }, []);
+
+  const handleDeleteComment = useCallback((commentId: number) => {
+    try {
+      setComments(prev => prev.filter(comment => comment.id !== commentId));
+      commentsService.deleteComment(commentId);
+    } catch (error) {
+      setIsError(true);
```

Comment:
- If you don't await the response you never catch an error so try/catch here doesn't make much sense
- Also use one style for API requests `try/catch` or with `then`